components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -2106,4 +2272,173 @@ private void handleInternalTokenRevocation(String consumerKey, Properties proper
+     * https://openid.net/specs/openid-connect-core-1_0.html#PairwiseAlg
+     *
+     * @param sectorIdentifierURI sector identifier URI
+     * @param redirectURIs        callBack URLs
+     * @throws IdentityOAuthClientException
+     */
+    private void validateSectorIdentifierURI(String sectorIdentifierURI, List<String> redirectURIs) throws
+            IdentityOAuthClientException {
+
+        if (StringUtils.isBlank(sectorIdentifierURI)) {
+            return;
+        }
+        URI uri = URI.create(sectorIdentifierURI);
+        String scheme = uri.getScheme();
+        if (StringUtils.isBlank(scheme) || !scheme.equals(String.valueOf(HttpsURL.DEFAULT_SCHEME))) {
+            throw handleClientError(INVALID_REQUEST, "Invalid sector identifier URI");
+        }
+        //validate whether sectorIdentifierURI points to JSON file containing an array of redirect_uri values
+        String validateSectorIdentifierURI = IdentityUtil.getProperty(
+                OAuthConstants.VALIDATE_SECTOR_IDENTIFIER);
+        if (StringUtils.isNotEmpty(validateSectorIdentifierURI) &&
+                Boolean.parseBoolean(validateSectorIdentifierURI)) {
+            try {
+                List<String> fetchedRedirectURI = new ArrayList<>();
+                ObjectMapper mapper = new ObjectMapper();
+                JsonNode redirectURIArray = mapper.readTree(uri.toURL());
+                if (redirectURIArray.isArray()) {
+                    Iterator<JsonNode> itr = redirectURIArray.iterator();
+                    while (itr.hasNext()) {
+                        JsonNode item = itr.next();
+                        fetchedRedirectURI.add(item.asText());
+                    }
+                }
+                if (!fetchedRedirectURI.containsAll(redirectURIs)) {
+                    throw handleClientError(INVALID_REQUEST, "Redirect URI missing in sector " +
+                            "identifier URI set");
+                }
+            } catch (IOException e) {
+                throw handleClientError(INVALID_REQUEST, "Invalid sector identifier URI");
+            }
+        }
+    }
+
+    /**
+     * Get call back URIs as a list
+     * @param application  OAuthConsumerAppDTO
+     * @return list of callback urls
+     */
+    private List<String> getRedirectURIList(OAuthConsumerAppDTO application) {
+
+        List<String> callBackURIList = new ArrayList<>();
+        // Need to split the redirect uris for validating the host names since it is combined
+        // into one regular expression.
+        if (application.getCallbackUrl().startsWith(
+                OAuthConstants.CALLBACK_URL_REGEXP_PREFIX)) {
+            String redirectURI = application.getCallbackUrl();
+            redirectURI = redirectURI.substring(redirectURI.indexOf("(") + 1,
+                    redirectURI.indexOf(")"));
+            callBackURIList = Arrays.asList(redirectURI.split("\\|"));
+        }
```

Comment:
The regex pattern could contain complete regexes can this logic handle that without breaking?